src/HtmlUtils.js

```diff
@@ -516,7 +516,10 @@ export function bodyToHtml(content, highlights, opts={}) {
         contentBodyTrimmed = contentBodyTrimmed.replace(ZWJ_REGEX, '');
 
         const match = EMOJI_REGEX.exec(contentBodyTrimmed);
-        emojiBody = match && match[0] && match[0].length === contentBodyTrimmed.length;
+        emojiBody = match && match[0] && match[0].length === contentBodyTrimmed.length
+                    // Prevent user pills expanding for users with only emoji in
+                    // their username
+                    && !content.formatted_body.includes("https://matrix.to/");
```

Comment:
probably need a `(content.formatted_body && !content.formatted_body.includes(...))` here just in case

Reply:
Ah, very good point.